### PR TITLE
Add subscription UIs for Cody services

### PR DESCRIPTION
**Pull request by eseliger**
This PR is quite a chunk, but it mainly consists of the following:
- Bring the codebase for the subscriptions pages into 2023. This includes design adjustments and moving to Apollo where possible, to make development easier. Also, IMO it looks nicer now.
- Implement the rate limit configuration work
- Change the mechanism for access tokens to "just serving" it for now. We don't have a proper access token system yet and will need to adjust a lot for that anyways and since the token is simply derived from the license key it's not more secret than the key itself, which we do show from the UI. This also allows us to create a self-serve experience.

### Screenshots

User viewing their subscription and self-serving the access token retrieval
Screenshot 2023-05-05 at 20 48 18@2x

Slightly redesigned subscriptions list
Screenshot 2023-05-05 at 20 49 02@2x

Redesigned subscription page with no license attached
Screenshot 2023-05-05 at 20 49 33@2x

New generate license modal based flow
Screenshot 2023-05-05 at 20 49 56@2x

Cody services after a license has been added, off by default
Screenshot 2023-05-05 at 20 50 12@2x

With completions toggled on
Screenshot 2023-05-05 at 20 50 35@2x

Create custom override for completions quota
Screenshot 2023-05-05 at 20 50 48@2x

Subscription with overridden quota
Screenshot 2023-05-05 at 20 51 03@2x

## Test plan

Lots of manual testing, this codebase is very old and doesn't have a ton of automated tests yet. I want to follow up here later and add proper storybook coverage.

### CODE REVIEW

File: enterprise/internal/licensing/llmproxy.go [hunk 1]
```diff
@@ -12,15 +12,18 @@ type LLMProxyRateLimit struct {
...
 
 	// TODO: Defaults for other plans
-
 	default:
-		return LLMProxyRateLimit{}
+		return LLMProxyRateLimit{
+			Limit:           10,
+			IntervalSeconds: 60 * 60 * 24,
+		}
 	}
```
Comment (eseliger) [inline enterprise/internal/licensing/llmproxy.go:28]: just added a little more here and made sure that the limit ain't 0 for other plans.

File: client/web/src/user/settings/UserSettingsSidebar.tsx [hunk 2]
```diff
@@ -74,7 +74,7 @@ export const UserSettingsSidebar: FC<UserSettingsSidebarProps> = props => {
                                     key={label}
                                     to={`/users/${user.username}/settings` + to}
                                     onClick={collapseMobileSidebar}
-                                    exact={true}
+                                    exact={exact}
```
Comment (eseliger) [inline client/web/src/user/settings/UserSettingsSidebar.tsx:77]: was broken.

File: migrations/frontend/1683246005_llmproxynoaccesstokenenable/up.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+ALTER TABLE product_licenses ALTER COLUMN access_token_enabled SET DEFAULT TRUE;
+-- One time migration to enable all licenses again, we want to disable them manually if abuse
+-- has been detected.
+UPDATE product_licenses SET access_token_enabled = TRUE;
```
Comment (bobheadxi) [inline migrations/frontend/1683246005_llmproxynoaccesstokenenable/up.sql:4]: at tad nervous about this, there are some many-years-old licenses out there 😬 what was the reasoning behind this again, instead of the existing migration + new licenses going forward?

File: enterprise/cmd/frontend/internal/dotcom/productsubscription/llmproxy_graphql_test.go [hunk 4]
```diff
@@ -50,11 +55,13 @@ func TestLLMProxyAccessResolverRateLimit(t *testing.T) {
...
 	t.Run("override default rate limit for a plan", func(t *testing.T) {
-		dbSubscriptions{db: db}.Update(ctx, subID, dbSubscriptionUpdate{
+		if err := (dbSubscriptions{db: db}.Update(ctx, subID, dbSubscriptionUpdate{
 			llmProxyAccess: &graphqlbackend.UpdateLLMProxyAccessInput{
 				RateLimit: pointify(int32(123456)),
 			},
-		})
+		})); err != nil {
+			t.Fatal(err)
+		}
```
Comment (bobheadxi) [inline enterprise/cmd/frontend/internal/dotcom/productsubscription/llmproxy_graphql_test.go:64]: nit: `require.NoError`

File: cmd/frontend/graphqlbackend/dotcom.graphql [hunk 5]
```diff
@@ -281,6 +265,11 @@ type ProductSubscription implements Node {
         first: Int
     ): ProductLicenseConnection!
     """
+    Access token for authenticating as the subscription holder with managed Sourcegraph services.
+    Null only if creating a token failed, for example when no active license exists.
+    """
+    sourcegraphAccessToken: String
+    """
```
Comment (bobheadxi) [inline cmd/frontend/graphqlbackend/dotcom.graphql:272]: How do you anticipate extending this to also show other active tokens? (as discussed, it would be useful for full sync) - concerned the naming is a bit wordy

**Review by bobheadxi — COMMENTED** [review]
looks awesome 😁 a few questions!

**Review by bobheadxi — COMMENTED** [review]
looks awesome 😁 a few questions!

**Review by bobheadxi — COMMENTED** [review]
looks awesome 😁 a few questions!

**Review by bobheadxi — COMMENTED** [review]
looks awesome 😁 a few questions!

**Review by bobheadxi — COMMENTED** [review]
looks awesome 😁 a few questions!

**Review by bobheadxi — COMMENTED** [review]
looks awesome 😁 a few questions!

**Comment by bobheadxi** [comment]
oopps.... hit refresh too many times when GitHub threw me a 500 😅

File: migrations/frontend/1683246005_llmproxynoaccesstokenenable/up.sql (hunk 3 above)
Comment (eseliger) [reply migrations/frontend/1683246005_llmproxynoaccesstokenenable/up.sql:4]: My understanding is that we will use llmproxy.enabled for access, and the old license key is merely a way to authenticate but not authorize. No strong opinion on this migration, happy to change it, but right now we don't have a way to enable/disable licenses from the UI so all on by default seemed the easiest until I get around adding a proper UI for listing historical keys and an option to disable them.

File: cmd/frontend/graphqlbackend/dotcom.graphql (hunk 5 above)
Comment (eseliger) [reply cmd/frontend/graphqlbackend/dotcom.graphql:272]: Yeah this predated the discussion on that a bit. 
I think we could simply make it `sourcegraphAccessTokens` and by default show the top one? Later, we can add a UI to show them all, and that API would also give you the endpoint you need for syncing the state from dotcom into LLM proxy. WDYT?

**Comment by eseliger** [comment]
> Generally I'd ask for 3 PRs based on the description 😂 but the screenshots LGTM

Yeah, sorry about that it was kind of in the flow that I discovered that the code is terribly old and rusty :( If you're concerned with that I can split it up further but that's gonna be a bit of extra work 😬

File: cmd/frontend/graphqlbackend/dotcom.graphql (hunk 5 above)
Comment (bobheadxi) [reply cmd/frontend/graphqlbackend/dotcom.graphql:272]: sounds good!

File: migrations/frontend/1683246005_llmproxynoaccesstokenenable/up.sql (hunk 3 above)
Comment (eseliger) [reply migrations/frontend/1683246005_llmproxynoaccesstokenenable/up.sql:4]: yeah makes sense. Let me default it to true for _new_ entries, but keep existing ones untouched since your migration already enabled active ones - does that make sense?

**Review by bobheadxi — APPROVED** [review]
